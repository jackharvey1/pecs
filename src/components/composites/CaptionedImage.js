import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Caption from '../atomics/Caption';
import Image from '../atomics/Image';

const InlineFlexDiv = styled.div`
    display: inline-flex;
    flex-direction: column;
    margin: 20px;
`;

const CaptionedImage = ({ id, src, alt, text, onDragStart }) => (
    <InlineFlexDiv>
        <Image
            id={id}
            src={src}
            alt={alt}
            onDragStart={onDragStart}
        />
        <Caption text={text} />
    </InlineFlexDiv>
);

CaptionedImage.propTypes = {
    src: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
    text: PropTypes.string,
    onDragStart: PropTypes.func.isRequired,
};

export default CaptionedImage;

