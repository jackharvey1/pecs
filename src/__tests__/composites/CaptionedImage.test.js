import React from 'react';
import renderer from 'react-test-renderer';
import CaptionedImage from '../../components/composites/CaptionedImage';
import { shallow } from 'enzyme';

describe('CaptionedImage component', () => {
    it('renders as intended', () => {
        const wrapper = renderer.create(<CaptionedImage
            id="container"
            src="http://image.com/image.png"
            alt="image"
            caption="Caption"
            index={0}
            onDragStart={() => null}
        />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('does not render caption when the id is not part of storyboard', () => {
        const wrapper = shallow(<CaptionedImage
            id="container"
            src="http://image.com/image.png"
            alt="image"
            caption="Caption"
            index={0}
            onDragStart={() => null}
        />);
        expect(wrapper.children().find('Caption').props().shouldDisplayCaption).toBe(false);
    });
});
