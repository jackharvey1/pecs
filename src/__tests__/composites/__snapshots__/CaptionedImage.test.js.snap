// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptionedImage component renders as intended 1`] = `
.c2 {
  color: #333333;
  font-size: 16px;
  font-family: Helvetica,Arial,sans-serif;
  margin-top: 12px;
  text-align: center;
}

.c1 {
  box-shadow: 0 2px 8px 0 rgba(0,0,0,.25);
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 20px;
}

<div
  className="c0"
>
  <img
    alt="image"
    className="c1"
    draggable={true}
    id="container"
    onDragStart={[Function]}
    src="http://image.com/image.png"
  />
  <p
    className="c2"
  >
    Caption
  </p>
</div>
`;
